compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaUnionTypeSymbol.java

```diff
@@ -70,7 +71,17 @@ public BallerinaUnionTypeSymbol(CompilerContext context, ModuleID moduleID, BFin
                 TypesFactory typesFactory = TypesFactory.getInstance(this.context);
 
                 for (BType memberType : ((BUnionType) this.getBType()).getMemberTypes()) {
-                    members.add(typesFactory.getTypeDescriptor(memberType));
+                    if (typesFactory.isTypeReference(memberType, memberType.tsymbol, false)
+                            || memberType.getKind() != TypeKind.FINITE) {
+                        members.add(typesFactory.getTypeDescriptor(memberType));
```

Comment:
Why do we have to check whether it's a type ref? Isn't checking whether it's not a finite type isn't enough? 🤔

Reply:
Let's say we have 
```ballerina
type T2 1|2|3;
type T4 "1"|"2"|T2;
```
If we get members of `T4` we get two `FiniteType`s (`T2` and `"1"|"2"|`). Therefore, we need to check whether `T2` is a type ref or not.